frontend/src/store/modules/product.js

```diff
@@ -1,27 +1,53 @@
-const product = {
+import axios from 'axios';
 
+const product = {
     namespaced: true,
-
     state: { 
-        newProduct: undefined
+        newProduct: undefined,
+        allProducts: [],
     },
 
     mutations: {
         setNewProduct (state, product) {
             state.newProduct = product
-        }
+            state.allProducts.push(product)
+        },
+        updateProducts(state, products) {
+            state.allProducts = products
+        },
+        removeProductByIndex(state, idx) {
+          delete state.allProducts[idx]
+        },
     },
-
     actions: { 
-        addNewProduct(context, product) {
-            context.commit('setNewProduct', product)
-        }
-    },
-    
+        addNewProduct({ commit }, product) {
+            // eslint-disable-next-line no-console
+            // axios.post({
+            //   url: 'https://localhost:5001/products/add', 
+            //   method: 'get',
+            //   headers: {'Access-Control-Allow-Origin': 'meeeee'},
+            //   data: product,
+            // })
+            axios.post('/api/products/add', product)
+            return commit('setNewProduct', product)
+        },
+        getAllProductsFromApi({ commit }) {
+            axios.get('/api/products').then(response => commit('updateProducts', response.data))
```

Comment:
This is _very_ well done. With ES6, you can change `.then`ables (as they're called) into `async/await` promise chains. I'll go over how to do this tomorrow!